### ISSUE TITLE
Support try/do blocks

### ISSUE THREAD

**Issue by BrianMWest** [phaazon/try-guard#1]
I've been finding this pretty useful to reduce some boilerplate and make things look just a little bit cleaner. One thing that would be helpful is this was expanded so that it works in try blocks instead of only functions:

One simple use case for an amethyst or specs project:
```
try {
    let entity = self.entity_opt?;
    guard!(world.is_alive(entity));
    world.delete_entity(entity).ok()?;
};
```

**Comment by phaazon** [comment]
Released as part of [try-guard-0.2.0](https://crates.io/crates/try-guard/0.2.0).

**Comment by BrianMWest** [comment]
Thanks for releasing that! Very excited to make use of it!
